content/1_docs/2_cookbook/7_extensions/0_subpage-builder/recipe.txt

```diff
@@ -148,3 +148,62 @@ function buildPageTree($page) {
 ```
 
 In this example, all subpages in the tree are automatically published. That way, they will be published together with the parent without any further interaction.
+
+## Creating subpages if the pages already exists
+
+The workflow we outlined above works with a `page.create:after` hook, so when a new page is created. But what if you want to restructure your site and add those subpages after (some) of the pages already exist.
+
+There are several ways to deal with this:
+
+### Using a `page.update:after` hook
+
+We can call the `buildPageTree()` method in a `page.update:after` hook to create the subpages when a page is updated.
+
+
+```php "/site/config/config.php"
+'hooks' => [
+    'page.update:after' => function ($newPage) {
+        if ($newPage->children()->isEmpty()) {
+          buildPageTree($newPage);
+        }
+    }
+]
+```
+
+Depending on whether the page can have other subpages, you might have to adapt the condition.
+
+In many cases, waiting for a page to be updated is not ideal, though.
+
+### Using a `route:after` hook
+
+We can use a `route.after` hook to call the `buildPageTree()` method when a page is visited rather then updated. This is usually more useful, because you would have to visit a page anyway to access the subpages:
+
+```php "/site/config/config.php"
+'hooks' => [
+    'route:after' => function ($route, $path, $method) {
+        $uid = explode('/', $path);
+        $uid = end($uid);
+        $uid = str_replace('+', '/', $uid);
+        $page = kirby()->page($uid);
+
+        if ($page->children()->isEmpty()) {
```

Comment:
I think this could lead to an error when `$page` is null/doesn't exist. Maybe add a `$page` check to the condition?

Reply:
How could I miss that, preaching that all the time in the forum 🙈 ?